### PR TITLE
Fix unit tests for new performance kernel

### PR DESCRIPTION
**Pull request by avbokovoy**
This PR addresses several issues, that are currently presented in kernel port:

- Added temporary unit test that assesses bwd kernel support range
- Fixed weighted mode
- Fixed wrong block and grid sizes

Command to launch new unit test (from fbgemm/test folder):
```shell
FBGEMM_TEST_WITH_ROCM=1 python -m tbe.training.backward_optimizers_test -k test_new_bwd_kernel
``` 
Current support range:

- **embeddings dimensions:** {64, 128, 192, 256}
- **weighted/unweighted:** supported
- **weight decay modes:** None, L2, DECOUPLE
- **nobag mode:** doesn't work properly (need to fix indexing)
- **optimizer:** rowwise_adagrad
- **mixed/mixed_B:** not supported
- **long segments:** supported

Unfortunately, previous unit tests that we used passed just because it didn't test new bwd kernel

### CODE REVIEW

**Review by xinyazhang — APPROVED** [review]
LGTM
